### DISCUSSION TITLE
high memory usage when open single source file

### DISCUSSION THREAD

**Discussion by katsusan** [golang/vscode-go#1882] [General]
I have heard that gopls will try to parse or load almost everything in memory, so not sure whether it is intended or beyond the design and post here before sumbiting a issue(similar issue: https://github.com/golang/vscode-go/issues/1859).

reproducing steps:
1. `ctrl + shift + N` to open a new window in VSCode.
2. open $GOROOT/src/cmd/compile/internal/ssa/rewritegeneric.go.
3. memory usage of gopls increased to about 2.9GB.

proof:
// open rewritegeneric.go   

![image](https://user-images.githubusercontent.com/11537821/140776289-3b84a5de-12c1-49ff-8864-ecdbb46c4e03.png)

// open rewritegeneric.go and src/cmd/compile/main.go   

![image](https://user-images.githubusercontent.com/11537821/140777742-03b6aa17-b928-410f-92f4-0783ea408fa5.png)

**Comment by hyangah** [comment]
Given the size of the Go project, I am not too surprised by the memory usage - Gopls basically parses and loads all the type information of files in the module even when opening a single file of a big module. Gopls team has some ideas to reduce the memory usage further. cc @findleyr
